src/litserve/api.py

```diff
@@ -50,8 +51,30 @@ def __init__(self, max_batch_size: int = 1, batch_timeout: float = 0.0, enable_a
         self.max_batch_size = max_batch_size
         self.batch_timeout = batch_timeout
         self.enable_async = enable_async
+
+        # Asyncify default decode_request if not implemented by user when enable_async is True
+        if self.enable_async and not asyncio.iscoroutinefunction(self.decode_request):
+            warnings.warn(
+                "'decode_request' is not implemented as an async coroutine. "
+                "Since 'enable_async=True', the default implementation will be wrapped to run asynchronously. "
+                "To avoid this warning and ensure proper async support, "
+                "please override 'decode_request' as 'async def decode_request(...)'.",
+                UserWarning,
+                stacklevel=2,
+            )
+            self.decode_request = self._asyncify(self.decode_request)
```

Comment:
needs to be applied to the encode_response as well which would be an async generator. We will have to run the generator in a thread and yield the output as soon as they are generated in that case. 

For predict, we can enforce users to define as async rather than asyncify.

Reply:
feel free to take a look at [this branch](https://github.com/Lightning-AI/LitServe/compare/asyncify?expand=1)

Reply:
Sure and thank you @aniketmaurya